src/providers/CollectionsProvider.tsx

```diff
@@ -0,0 +1,167 @@
+'use client';
+
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+import { Invocation } from './InvocationProvider';
+
+export type Folder = {
+	id: string;
+	name: string;
+	invocations: Invocation[];
+};
+
+type Collection = {
+	id: string;
+	name: string;
+	folders: Folder[];
+};
+/* eslint-disable @typescript-eslint/no-empty-function */
+const CollectionsContext = createContext<{
+	loading: boolean;
+	collections: Collection[];
+	addCollection: (name: string) => void;
+	removeCollection: (id: string) => void;
+	selectedCollection: Collection | null;
+	selectCollection: (id: string) => void;
+	addFolderToCollection: (id: string, name: string) => void;
+	removeFolderFromCollection: (id: string, folderId: string) => void;
+}>({
+	loading: true,
+	collections: [],
+	addCollection: () => {},
+	removeCollection: () => {},
+	selectedCollection: null,
+	selectCollection: () => {},
+	addFolderToCollection: () => {},
+	removeFolderFromCollection: () => {},
+});
+
+export const CollectionsProvider = ({
+	children,
+}: {
+	children: React.ReactNode;
+}) => {
+	const [loading, setLoading] = useState(true);
+	const [collections, setCollections] = useState<Collection[]>([]);
+	const [selectedCollection, setSelectedCollection] =
+		useState<Collection | null>(null);
+
+	useEffect(() => {
+		const localData = localStorage.getItem('collections');
+		const newValue = localData ? JSON.parse(localData) : [];
+		setCollections(localData ? newValue : []);
```

Comment:
maybe this will be better

```
const collectionsFromLocalStorage = localStorage.getItem('collections');
 const collectionsData = collectionsFromLocalStorage ? JSON.parse(collectionsFromLocalStorage) : []; setCollections(collectionsData);
```